### PR TITLE
Fixed Store Response headers case insensitivity

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Fixed Store Response headers case insensitivity
* Response headers are case insensitive, and they can change the casing anytime. 
* FYI - This was the PR where case insensitivity was removed from `StoreResponse` headers - https://github.com/Azure/azure-sdk-for-java/pull/28620 
* While using maps instead of arrays as headers is still important, it can be done using some implementation of `SortedMap` with comparator implemented for case insensitivity. This will be done in future PRs. 
* Created this github issue to track it - https://github.com/Azure/azure-sdk-for-java/issues/29206

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreClient.java [hunk 1]
```diff
@@ -152,10 +152,20 @@ private RxDocumentServiceResponse completeResponse(
         StoreResponse storeResponse,
         RxDocumentServiceRequest request) throws InternalServerErrorException {
 
-        Map<String, String> responseHeaders = new HashMap<>(storeResponse.getResponseHeaders());
+        if (storeResponse.getResponseHeaderNames().length != storeResponse.getResponseHeaderValues().length) {
+            throw new InternalServerErrorException(RMResources.InvalidBackendResponse);
+        }
+
+        Map<String, String> headers = new HashMap<>(storeResponse.getResponseHeaderNames().length);
+        for (int idx = 0; idx < storeResponse.getResponseHeaderNames().length; idx++) {
```
[the remark this answers is not in the document]
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreClient.java:160]: That's what I plan to do, but as part of the above linked github issue future PR. 
In this PR, I just reverted to the old state of v4 SDK, so as to make sure if compute or gateway changes anything, we don't break existing customers.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java [hunk 2]
```diff
@@ -35,11 +36,20 @@ public class StoreResponse {
 
     public StoreResponse(
             int status,
-            Map<String, String> headerMap,
+            List<Map.Entry<String, String>> headerEntries,
             byte[] content) {
 
         requestTimeline = RequestTimeline.empty();
-        responseHeaders = new ConcurrentHashMap<>(headerMap);
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java:42]: Why not just use java.util.concurrent.ConcurrentSkipListMap here instead?

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]

**Comment by FabianMeiswinkel** [comment]
Here is an alternative which does not fallback to arrays - https://github.com/Azure/azure-sdk-for-java/pull/29268

**Comment by xinlian12** [comment]
close due to PR https://github.com/Azure/azure-sdk-for-java/pull/29268

**Pull request closed without merging**